pkg/asset/installconfig/openstack/validation/platform.go

```diff
@@ -19,8 +19,8 @@ func ValidatePlatform(p *openstack.Platform, n *types.Networking, ci *CloudInfo)
 	var allErrs field.ErrorList
 	fldPath := field.NewPath("platform", "openstack")
 
-	// validate BYO machinesSubnet usage
-	allErrs = append(allErrs, validateMachinesSubnet(p, n, ci, fldPath)...)
+	// validate BYO controlPlanePort usage
```

Comment:
Adding a check to see if ControlPlanePort config will avoid a calll to validate if the config is not present.
```
if p.ControlPlanePort != nil {
        allErrs = append(allErrs, validateControlPlanePort(p, n, ci, fldPath)...)
}
```

Reply:
I was trying to follow the pattern of the other validate functions. Is the intention to move this check out of the `validateControlPlanePort` func to reduce the numbers indentations present at `validateControlPlanePort`?

Reply:
I added a check for existence of `ControlPlanePort`. Regarding the ExternelDNS, if it would be present and we skip the validation, no error would be displayed as no validation would happen, I think it would be better to avoid that.
Let me know if you have a different thought.